features/attach_validtoken.feature

```diff
@@ -0,0 +1,31 @@
+Feature: Command behaviour when attaching a machine to an Ubuntu Advantage
+        subscription using a valid token
+
+    Scenario: Attach command in a trusty lxd container
+       Given a trusty lxd container with ubuntu-advantage-tools installed
+        When I attach "<contract_token>" with sudo
+        Then stdout will include:
+        """
+        ESM Infra enabled
+        """
+        And stdout will include:
+        """
+        This machine is now attached to
+        """
+        And stdout will include:
+        """
+        SERVICE       ENTITLED  STATUS    DESCRIPTION
+        cc-eal        yes       n/a       Common Criteria EAL2 Provisioning Packages
```

Comment:
As you mentioned you are dealing with non-printable control characters being turned into spaces offsetting the stdout alignment for the cc-eal  "yes" and "n.a" columns when compare context.process.stdout to the properly aligned context.text in your "stdout will include" assertion.

One minimal solution could have been to just copy and paste the failure output text you see and replace the misaligned text including those non-printables notice the n/a doesn't align with STATUS header

```
          """
          SERVICE       ENTITLED  STATUS    DESCRIPTION
          cc-eal        yes                n/a                Common Criteria EAL2 Provisioning Packages
          cis-audit     no                 —                  Center for Internet Security Audit Tools
          esm-apps      no                 —                  UA Apps: Extended Security Maintenance
          esm-infra     yes                enabled            UA Infra: Extended Security Maintenance
          fips          yes                n/a                NIST-certified FIPS modules
          fips-updates  yes                n/a                Uncertified security updates to FIPS modules
          livepatch     yes                n/a                Canonical Livepatch service
          """        
```
The above suggestion will get you a pass because it'll be the exact misaligned content that behave was complaining about.

Better alternative solution:

I think you really need a regex "then" step

So this step should be the following: use the + to indicate greater than one space
```suggestion
And stdout matches regexp:
      """
      SERVICE       ENTITLED  STATUS    DESCRIPTION
      cc-eal       +yes      +n/a      +Common Criteria EAL2 Provisioning Packages
      cis-audit    +no       +—        +Center for Internet Security Audit Tools
      esm-apps     +no       +—        +UA Apps: Extended Security Maintenance
      esm-infra    +yes      +enabled  +UA Infra: Extended Security Maintenance
      fips         +yes      +n/a      +NIST-certified FIPS modules
      fips-updates +yes      +n/a      +Uncertified security updates to FIPS modules
      livepatch    +yes      +n/a      +Canonical Livepatch service
      """
```
Your new @then step is commented below